scripts/valhalla_build_elevation

```diff
@@ -1,81 +1,249 @@
+    max_x, max_y = [ceil(x) for x in bounds[-2:]]
+    for x in range(min_x, max_x):
+        for y in range(min_y, max_y):
+            yield [x, y, x + 1, y + 1]
+
+
+def get_tiles_with_geojson(input_dir: Path) -> List[Tile]:
+    try:
+        from shapely.geometry import Polygon, box
+    except ImportError:
+        LOGGER.critical(
+            "Could not import shapely. Please install shapely or use another download method instead."
+        )
+        sys.exit(1)
+
+    if not input_dir.is_dir() or not len(list(input_dir.glob('*.geojson'))) > 0:
+        LOGGER.critical(f"Geojson directory does not exist or contains no GeoJSON files: {input_dir.resolve()}")
+        sys.exit(1)
+
+    def get_outer_rings(input_dir: Path) -> Iterable[Polygon]:
+        for file in input_dir.glob("*.geojson"):
+            with open(file) as f:
+                geojson = json.load(f)
+            for feature in geojson["features"]:
+                if feature["geometry"]["type"] == "Polygon":
+                    yield Polygon(feature["geometry"]["coordinates"][0])
+                if feature["geometry"]["type"] == "MultiPolygon":
+                    for single_polygon in feature["geometry"]["coordinates"]:
+                        yield Polygon(single_polygon[0])
+
+    tile_infos = []
+    for poly in get_outer_rings(input_dir):
+        for rect in grid_from_bounds(poly.bounds):
+            if poly.intersects(box(*rect)):
+                s, w, *_ = rect
+                tile_infos.append(get_tile_info(s, w))
+
+    return tile_infos
+
+
+def get_tiles_with_bbox(bbox_str: str) -> List[Tile]:
+    try:
+        bbox = min_x, min_y, max_x, max_y = [float(x) for x in bbox_str.split(",")]
+    except ValueError:
+        LOGGER.critical(f"BBOX {bbox_str} is not a comma-separated string of coordinates.")
+        sys.exit(1)
+
+    # validate bbox
+    if (
+        min_x > max_x
+        or min_x < -180
+        or max_x > 180
+        or min_y > max_y
+        or min_y < -90
+        or max_y > 90
+    ):
+        LOGGER.critical(f"Bbox invalid: {bbox}")
+        sys.exit(1)
+
+    LOGGER.debug(f"Received valid bbox: {bbox}")
+
+    tile_infos = []
+    for grid in grid_from_bounds(bbox):
+        s, w = grid[:2]
+        tile_infos.append(get_tile_info(s, w))
+
+    return tile_infos
+
+
+def get_tiles_with_graph(graph_dir: Path) -> List[Tile]:
+    if not graph_dir.is_dir():
+        LOGGER.critical(f"Graph directory {graph_dir.resolve()} does not exist.")
+        sys.exit(1)
+
+    tiles = []
+    local_dir = graph_dir.joinpath('1')
+    tile_width = int(360 / LOCAL_SIZE)
+    for tile_fp in local_dir.rglob('*.gph'):
+        # turn the path into a tile ID
+        tile_id = int(str(tile_fp.parent.relative_to(local_dir).joinpath(tile_fp.stem)).replace(os.sep, ''))
+        s, w = (tile_id % tile_width) * LOCAL_SIZE - 180, int(tile_id / tile_width) * LOCAL_SIZE - 90
+
+        tiles.append(get_tile_info(s, w))
+
+    return tiles
```

Comment:
i love that someone is finally implementing this idea however this doesnt look right to me. for this to work you need to turn the gph tile into a bounding box and see which srtm tiles it intersects.to mee it looks like you just turn the tile id into its index, thats a good start, but you need to take the tile level into account because that determins what size it is. you cannot just look at at level 1 tiles because they only hold middle class roads. you will be missing places most likely. do you want me to share a snippet?

Reply:
ah yeah I did mess smth up in my brain.. somehow my line of thought was: if there's a 2nd level tile, there'll definitely be the 1st level tile. but that's not correct of course (but the other way around is correct: there'll always have to be a 2nd level tile if there's features in any other level, for the bins). other than that the logic is correct, we don't need really a bbox in this case bcs 1st level graph and elevation tiles have the same dimension and conventions, so turning the tile ID into a lower-left corner is actually enough to get the correct elevation tile.

I'll turn it into 2nd level logic.